docs/reference/indices/rollover-index.asciidoc

```diff
@@ -43,530 +30,396 @@ POST /alias1/_rollover/my-index-000002
+`rolled_over`::
+(Boolean)
+If `true`, the data stream or index alias rolled over.
 
-include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=settings]
+`dry_run`::
+(Boolean)
+If `true`, {es} did not perform the rollover.
 
+`condition`::
+(object)
+Result of each condition specified in the request's `conditions`. If no
+conditions were specified, this is an empty object.
++
+.Properties of `condition`
+[%collapsible%open]
+====
+`<condition>`::
+(Boolean) The key is each condition. The value is its result. If `true`, the
+index met the condition at rollover.
+====
 
 [[rollover-index-api-example]]
 ==== {api-examples-title}
 
-[[rollover-index-basic-ex]]
-===== Basic example
-
-[source,console]
---------------------------------------------------
-PUT /logs-000001 <1>
-{
-  "aliases": {
-    "logs_write": {}
-  }
-}
-
-# Add > 1000 documents to logs-000001
-
-POST /logs_write/_rollover <2>
-{
-  "conditions": {
-    "max_age":   "7d",
-    "max_docs":  1000,
-    "max_size":  "5gb",
-    "max_primary_shard_size": "2gb"
-  }
-}
---------------------------------------------------
-// TEST[setup:my_index_huge]
-// TEST[s/# Add > 1000 documents to logs-000001/POST _reindex?refresh\n{"source":{"index":"my-index-000001"},"dest":{"index":"logs-000001"}}/]
-<1> Creates an index called `logs-0000001` with the alias `logs_write`.
-<2> If the index pointed to by `logs_write` was created 7 or more days ago, or
-    contains 1,000 or more documents, or has an index size at least around 5GB, then the `logs-000002` index is created
-    and the `logs_write` alias is updated to point to `logs-000002`.
-
-The API returns the following response:
+[[roll-over-data-stream]]
+===== Roll over a data stream
 
-[source,console-result]
---------------------------------------------------
-{
-  "acknowledged": true,
-  "shards_acknowledged": true,
-  "old_index": "logs-000001",
-  "new_index": "logs-000002",
-  "rolled_over": true, <1>
-  "dry_run": false, <2>
-  "conditions": { <3>
-    "[max_age: 7d]": false,
-    "[max_docs: 1000]": true,
-    "[max_size: 5gb]": false,
-    "[max_primary_shard_size: 2gb]": false
-  }
-}
---------------------------------------------------
+The following request rolls over a data stream unconditionally.
 
-<1> Whether the index was rolled over.
-<2> Whether the rollover was dry run.
-<3> The result of each condition.
+[source,console]
+----
+POST my-data-stream/_rollover
+----
+// TEST[setup:my_data_stream]
 
+:target: data stream
+:index: write index
 
-[[rollover-data-stream-ex]]
-===== Roll over a data stream
+// tag::rollover-conditions-ex[]
+The following request only rolls over the {target} if the current {index}
+meets one or more of the following conditions:
 
-[source,console]
------------------------------------
-PUT _index_template/template
-{
-  "index_patterns": ["my-data-stream*"],
-  "data_stream": { }
-}
------------------------------------
+* The index was created 7 or more days ago.
```

Comment:
There's no description of the `max_primary_shard_size` option. We might also want to remove the `max_size` option as it is being deprecated in favor of the new `max_primary_shard_size` option.

Reply:
I reuse the `conditions` options from the ILM rollover action. It's not easily viewed in the source, but you can see the section in the preview:

https://elasticsearch_70938.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html#rollover-index-api-request-body

<img width="683" alt="Screen Shot 2021-03-30 at 9 58 07 AM" src="https://user-images.githubusercontent.com/40268737/113001077-714fff00-913e-11eb-997f-322d5e84f5ce.PNG">

Reply:
I'll also leave in the `max_size` option for now. When we officially deprecate it, I'll add the appropriate admon.